File: Frontend/src/interfaces/v12/product/productCreateUpdate/productCreateUpdateDto.ts
import { guid } from "types/guid";

export interface ProductCreateUpdateDto {
  productCode: string;

  // Foreign keys to Lookup table
  categoryId: guid;
  brandId?: guid;
  model?: string;
  year?: number;
  colorId?: guid;
  storageId?: guid;
  ramId?: guid;
  processorId?: guid;
  screenSizeId?: guid;
  gradeId?: guid;

  lowQty: number;
}

File: Wms.Api/Dto/Product/ProductCreateUpdate/ProductCreateUpdateDto.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Wms.Api.Dto.Product.ProductCreateUpdate
{
    public class ProductCreateUpdateDto
    {
        [Required]
        public string ProductCode { get; set; } = default!;
        
        // Foreign keys to Lookup table
        [Required]
        public Guid CategoryId { get; set; }
        public Guid? BrandId { get; set; }
        public string? Model { get; set; }
        public Guid? ColorId { get; set; }
        public Guid? StorageId { get; set; }
        public Guid? RamId { get; set; }
        public Guid? ProcessorId { get; set; }
        public Guid? ScreenSizeId { get; set; }

        public int LowQty { get; set; } = 0;
    }
}
